### ISSUE TITLE
Internal Error: Illegal Access 

### ISSUE THREAD

**Issue by griffinmacias** [node-inspector/node-inspector#864]
Node version 6.0.0
Node Inspector 0.12.8
Express 4.13.4 

![screen shot 2016-05-05 at 9 37 54 am](https://cloud.githubusercontent.com/assets/11699932/15044831/212704e6-12a5-11e6-82dd-c48441e65cd5.png)

1) node-debug app.js
2) open the inspector and press play
3) open the localhost:3000
4) works then set breakpoint on 'res.send() line'; 
5) reload localhost:3000
6) internal error: illegal access

I am aware of the issue #413 and my issue might be related. Why am I getting this error, I am not able to set a breakpoint anywhere.

**Comment by LeandroFavero** [comment]
Same problem, and happens only with Node 6.0

**Comment by chippawah** [comment]
Same here.

**Comment by theHazzard** [comment]
Same Problem

**Comment by MKHenson** [comment]
Same here

**Comment by wokalski** [comment]
I have this issue only if I try to debug ES6 code. I run `node-inspector` and then invoking `babel-node debug -- myscript.js`. It breaks on the first line of a script and everything seems fine, but then it hangs and after some times outputs the error message.

Do you know any workarounds?

---
Sorry for any misleading thoughts. I'm completely new to the whole js stack.

**Comment by rcosnita** [comment]
As a work around you can use --debug-brk argument which will enable a breakpoint on the first line of the script you run. Your debug session will work as expected.

**Comment by griffinmacias** [comment]
@rcosnita I tried that, still didn't work.

**Comment by griffinmacias** [comment]
Thanks @rcosnita, that works

**Comment by theHazzard** [comment]
it didn't work for me
every time that I try to debug inside a callback from requestjs it throws Illegal Access

**Comment by bmcswee** [comment]
Also not working for me - already using --debug-brk.

Error gets thrown no matter where I put my breakpoint.

**Comment by alfg** [comment]
Same problem here. Node 6+ when setting breakpoint on a callback.

**Comment by jaredwy** [comment]
As part of my work i've had to track down what was causing this issue in our setup (not node, but we use node inspector). The culmination of that investigation was https://bugs.chromium.org/p/v8/issues/detail?id=5010.The symptoms were the same, and as @rcosnita they seemed to improved when using --debug-brk in some cases. So i suspect they are likely the same, im going to start looking to fix this on monday.

**Comment by blake-regalia** [comment]
Same error here on node v6.1.0 & node-inspector v0.12.8; using --debug-break

As @alfg noted, it only happens (for me at least) when the breakpoint is within a callback that is reached after a different breakpoint outside the callback is encountered. Trying now to make a reproducible example

**Comment by blake-regalia** [comment]
Got it. It happens when trying to inspect a variable that is the result from calling `bind` on a function:

```js
let crash_inspector = (function(){}).bind({});
debugger;
```

To prove that inspecting it is the culprit, you can see it happen like this:
```js
(() => {
	let try_to_inspect_this_variable = {
		crash_inspector: (function(){}).bind({}),
	};
	debugger;
})();
```

**Comment by jaredwy** [comment]
Yup. I've got a patch in progress for v8. Will update this issue as it lands.

**Comment by theHazzard** [comment]
Hi! 

it always throws while debugging inside a `request` callback
```javascript
const request = require('request');

request.get('http://www.google.com', (err, res, body) => {
    debugger;
});
```

**Comment by eugenioclrc** [comment]
+1

**Comment by blake-regalia** [comment]
Fixed with the latest node version 6.2.0
https://github.com/nodejs/node/pull/6572

**Comment by blake-regalia** [comment]
This is a v8 problem that was addressed with this commit:
https://github.com/v8/v8/commit/246d5bba8f8eff204449ecc183a486b9c7722ba5

Now you'll have to wait until this v8 dep appears in a node.js upgrade

**Comment by jaredwy** [comment]
FYI I have informed node of the problem.

**Comment by LeandroFavero** [comment]
I updated to 6.2.0, but there are other issue:

Cannot find module '/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.7.3/node-v48-darwin-x64/debug.node'

**Comment by wokalski** [comment]
@LeandroFavero remove `node_modules` and run `npm install`. (Or just remove the `node-inspector` module)

**Comment by superclarkk** [comment]
@wokalski That is not the answer. The v8 fix has not been integrated into node yet.

**Comment by theHazzard** [comment]
I'm still having the same issue with node 6.2, illegal access while
debugging inside of a request callback :(

*Enzo Luis Stróngoli*
*NodeJS Developer*
*http://www.turismocity.com.ar <http://www.turismocity.com.ar/>*
*https://www.linkedin.com/in/enzostrongoli
<https://www.linkedin.com/in/enzostrongoli>*
*strongolienzo@gmail.com* <strongolienzo@gmail.com>
*enzo@turismocity.com <enzo@turismocity.com>*

2016-05-19 12:02 GMT-03:00 Wojtek Czekalski <notifications@github.com>:

**Comment by wokalski** [comment]
@superclarkk It is not a fix for illegal access, but it is a fix for @LeandroFavero issue, as far as I know.

**Comment by theHazzard** [comment]
sorry, I thought that was for this issue

*Enzo Luis Stróngoli*
*NodeJS Developer*
*http://www.turismocity.com.ar <http://www.turismocity.com.ar/>*
*https://www.linkedin.com/in/enzostrongoli
<https://www.linkedin.com/in/enzostrongoli>*
*strongolienzo@gmail.com* <strongolienzo@gmail.com>
*enzo@turismocity.com <enzo@turismocity.com>*

2016-05-19 14:30 GMT-03:00 Wojtek Czekalski <notifications@github.com>:

**Comment by superclarkk** [comment]
@wokalski I have the same issue as @LeandroFavero and also the illegal access issue (I presume they are connected). Deleting and reinstalling all modules does not fix it.

**Comment by wokalski** [comment]
@superclarkk No, they are not connected afaik. I had this issue because I installed `node-inspector` with some version of `node` and used it with other. I'm sorry if my tip is misleading!

**Comment by nicobevilacqua** [comment]
Same problem here with node 6.2 and node-inspector 0.12.8

**Comment by jaredwy** [comment]
https://github.com/nodejs/node/pull/6928 is up that will include the v8 fix that is required to fix this.

**Comment by sunilalgos** [comment]
Am also facing same problem with node 6.2.0 and node-inspector 0.12.8 version. This is not working with --debug and -debug-brk flag too. Are there any alternatives in solving this problem like using earlier versions like so ? If possible please update me.

**Comment by kunal-mandalia** [comment]
Same as a bunch of other folk. Node 6.2.0, node-inspector 0.12.8. 
Getting 'Internal error: illegal access' after running 'node-inspector' and 'node --debug-brk index.js' (or node --debug index.js) and setting a breakpoint on the following code:

```
var express = require('express');
var app = express();

app.get('/', function (req, res) {
  res.send('Hello World!'); <-- breakpoint set here
});

app.listen(3003, function () {
  console.log('Example app listening on port 3003!');
});
```
Hope the JavaScript Gods heed the call of the humble debuggers!

**Comment by mgol** [comment]
Node 6.2.1 has been released and is supposed to fix this issue.

**Comment by kunal-mandalia** [comment]
I can confirm node 6.2.1 let's node-inspector work as expected 👍

**Comment by rohitsi** [comment]
After updating to node 6.2.1, was able to debug a strongloop app... Great work guys. Thanks a ton!

**Comment by aa6** [comment]
On node v6.2.1 I still have a problem debugging this code:
```javascript
var Human = function(name)
{
    this.name = name
}
Human.prototype.speak = function(cb)
{
    console.log("My name is " + this.name)
}

var humanHandler =
{
    apply: function(target, this_arg, args) 
    {
        return new (Function.prototype.bind.apply(Human, [null].concat(Array.prototype.slice.call(args)))) 
    }
}

var ProxyHuman = new Proxy(function(){},humanHandler)

var tony = ProxyHuman("Tony")
tony.speak()
```
It gives `Internal error: illegal access` in the debugger if I resume script execution.
```console
me@me:~/tst/openca$ node --version
v6.2.1
me@me:~/tst/openca$ node-inspector --version
Node Inspector v0.12.8
me@me:~/tst/openca$ node bug.js
My name is Tony
me@me:~/tst/openca$ node-debug bug.js
Node Inspector v0.12.8
Visit http://127.0.0.1:8080/?port=5858 to start debugging.
Debugging `bug.js`

Debugger listening on port 5858
```

**Comment by iyobo** [comment]
Just verified that this is be resolved in node 7.7+
